incident_viewer_demo/src/LineChart/LineChart.jsx

```diff
@@ -305,37 +316,86 @@ class LineChart extends PureComponent {
+          .attr('height', plotAreaHeight-goodYmax);
+
+          if (rectFitsText) {
+            this.infoHoverBox.append('text')
+            .classed('good-hover-text', true)
+            .attr('x', xScale(goodIncidentSeries.start) + 0.5*width)
+            .attr('y', goodYmax + verticalTextShifter*(plotAreaHeight-goodYmax))
+            .append('svg:tspan')
+            .attr('x', xScale(goodIncidentSeries.start) + 20)
+            .attr('dy', 0)
+            .text(goodDescription1)
+            .append('svg:tspan')
+            .attr('x', xScale(goodIncidentSeries.start) + 20)
+            .attr('dy', 20)
+            .text(goodDescription2)
+          }
+        }
+
+        // Draw the hover state for the bad period information
+        if (highlightedDate.isSameOrBefore(badIncidentSeries.end) && highlightedDate.isSameOrAfter(badIncidentSeries.start) && mouseY > badYmax) {
+          this.infoHoverBox.append('rect')
+          .classed("bad-incident-area", true)
+          .attr('x', xScale(badIncidentSeries.start))
+          .attr('y', badYmax)
+          .attr('width', width)
+          .attr('height', plotAreaHeight-badYmax)
+          
+          if (rectFitsText) {
+            this.infoHoverBox.append('text')
+            .classed('bad-hover-text', true)
+            .attr('x', xScale(badIncidentSeries.start) + 0.5*width)
+            .attr('y', badYmax + verticalTextShifter*(plotAreaHeight-badYmax))
+            .append('svg:tspan')
+            .attr('x', xScale(badIncidentSeries.start) + 20)
+            .attr('dy', 0)
+            .text(goodDescription1)
+            .append('svg:tspan')
+            .attr('x', xScale(badIncidentSeries.start) + 20)
+            .attr('dy', 20)
+            .text(goodDescription2)
+          }
+        }
+
+        // Draw the hover state for the incident information
+        if (highlightedDate.isSameOrBefore(badIncidentSeries.end) && highlightedDate.isSameOrAfter(badIncidentSeries.start) && mouseY < badYmax && mouseY > goodYmax) {
+          this.infoHoverBox.append('rect')
+          .classed("incident-area", true)
+          .attr('x', xScale(badIncidentSeries.start))
+          .attr('y', goodYmax)
+          .attr('width', xScale(badIncidentSeries.end) - xScale(badIncidentSeries.start))
+          .attr('height', badYmax-goodYmax)
+
+          if (rectFitsText) {
+            this.infoHoverBox.append('text')
+            .classed('incident-hover-text', true)
+            .attr('x', xScale(badIncidentSeries.start) + 0.5*width)
+            .attr('y', goodYmax + verticalTextShifter*(plotAreaHeight-goodYmax - badYmax/2))
+            .append('svg:tspan')
+            .attr('x', xScale(badIncidentSeries.start) + 20)
+            .attr('dy', 0)
+            .text(goodDescription1)
+            .append('svg:tspan')
+            .attr('x', xScale(badIncidentSeries.start) + 20)
+            .attr('dy', 20)
+            .text(goodDescription2)
+          }
+        }
```

Comment:
Would it make sense to modularize this code for the hover states? This is a big chunk of mostly repeated code.